volumina/pixelpipeline/imagesources.py

```diff
@@ -420,8 +423,95 @@ def toImage( self ):
                 #FIXME: maybe this should be done in a better way using an operator before the colortable request which properly handles 
                 #this problem 
                 warnings.warn("Data for colortable layers cannot be float, casting",RuntimeWarning)
-                a=a.astype(np.int32)
-            vigra.colors.applyColortable(a, self._colorTable, byte_view(img))
+                a = np.asanyarray(a, dtype=np.uint32)
+
+            # If we have a masked array with a non-trivial mask, ensure that mask is made transparent.
+            _colorTable = self._colorTable
+            if np.ma.is_masked(a):
```

Comment:
I was thinking we could have a utility function that ensures there is a transparent layer as the first element of a colortable (adding one if it is missing). Where would you like me to put this utility function?